### DISCUSSION TITLE
Headless Linux Support?

### DISCUSSION THREAD

**Discussion by afinetapestry** [dotnet/Silk.NET#730] [Q&A] [answered]
Hey everyone,

I'm trying to put together an OpenGL Dotnet Core app that runs on headless Linux (AWS EC2/nvidia Tesla although that shouldn't matter) and Silk.NET looks like it could be a good choice.

[This old blog from nvidia](https://developer.nvidia.com/blog/egl-eye-opengl-visualization-without-x-server/) suggests that I need to call a few EGL functions (`eglInitialize()`, `eglChooseConfig()`, `eglCreateContext()`, ...) to use EGL without X11 running.

Is EGL supported in Silk.NET? I couldn't find any references to the required functions in the repo. Does [Silk.NET.EGL](https://www.nuget.org/packages/Silk.NET.EGL/) live in a separate repo? I couldn't find an API reference for it either.

Thanks and Merry Christmas,
Jon

**Comment by Perksey** [comment, marked as answer]
We use GLFW and SDL under-the-hood. Supposedly, there's a way you can get GLFW to use EGL.

On GLFW I found this: [GLFW_CONTEXT_CREATION_API](https://www.glfw.org/docs/latest/window_guide.html#window_hints_ctx) which I will promote to an issue to expose this hint.

We silently dropped our official EGL bindings during the 2.0 transition because we weren't using them and neither were our users, so it was unnecessary maintenance cost. We don't intend to bring them back until at least 3.0 (currently in the design stage, and not in active development yet)

For now, try using this workaround:
```cs
GlfwWindowing.Use();
GlfwProvider.GLFW.Value.WindowHint(WindowHintContextApi.ContextCreationApi, ContextApi.EglContextApi);
var window = Window.Create(WindowOptions.Default with { ... });
```

**Reply by afinetapestry** [reply]
Hi Dylan, Thanks for your quick response.

Due to a number of factors we've decided to C++ for this project rather than C# and implement the required libraries directly.

**Reply by Perksey** [reply]
Understandable!
